### PR TITLE
chore(deps): update to Quarkus 3.17.0.CR1

### PR DESCRIPTION
**Pull request by metacosm**
Signed-off-by: Chris Laprun <claprun@redhat.com>

### CODE REVIEW

File: bundle-generator/deployment/src/main/java/io/quarkiverse/operatorsdk/bundle/deployment/BundleProcessor.java [hunk 1]
```diff
@@ -121,7 +121,7 @@ CSVMetadataBuildItem gatherCSVMetadata(KubernetesConfig kubernetesConfig,
 
         // note that version, replaces, etc. should probably be settable at the reconciler level
         // use version specified in bundle configuration, if not use the one extracted from the project, if available
-        final var version = kubernetesConfig.getVersion().orElse(appConfiguration.getVersion());
+        final var version = kubernetesConfig.version().orElse(appConfiguration.getVersion());
```
Comment (xstefank) [inline bundle-generator/deployment/src/main/java/io/quarkiverse/operatorsdk/bundle/deployment/BundleProcessor.java:124]: Not sure if we should push different changes in the update commits.

File: pom.xml [hunk 2]
```diff
@@ -79,6 +79,7 @@
       <properties>
         <skipTests>true</skipTests>
         <skipITs>true</skipITs>
+        <skipDocs>true</skipDocs>
```
[the remark this answers is not in the document]
Comment (metacosm) [reply pom.xml:82]: Weird…

File: bundle-generator/deployment/src/main/java/io/quarkiverse/operatorsdk/bundle/deployment/BundleProcessor.java (hunk 1 above)
Comment (metacosm) [reply bundle-generator/deployment/src/main/java/io/quarkiverse/operatorsdk/bundle/deployment/BundleProcessor.java:124]: This is a required change. The code won't compile without this change. This is the reason why the ecosystem checks have been failing for so long as well.

Comment (xstefank) [reply bundle-generator/deployment/src/main/java/io/quarkiverse/operatorsdk/bundle/deployment/BundleProcessor.java:124]: OK, thanks for the explanation.

**Pull request merged**